### DISCUSSION TITLE
Upgrading from 2.7.8 to 3

### DISCUSSION THREAD

**Discussion by n3bojs4** [netbox-community/netbox#7086] [Q&A] [answered]
Hello,

Please can you tell me what's the best practice for upgrading. I am running a 2.7.8 with python 3.5.3.
Do i need to upgrade my python and go to version 2.8 first before going to 2.11 and to v3 ?
Thanks for your advice !

Regards

**Comment by sdktr** [comment, marked as answer]
You should be fine with two steps:
1) 2.11 with python 3.7+
2) 3.0 with python 3.7+

BUT: always BACKUP your database first PLUS test a restore

**Reply by n3bojs4** [reply]
> 
> 
> You should be fine with two steps:
> 
>     1. 2.11 with python 3.7+
> 
>     2. 3.0 with python 3.7+
> 
> 
> BUT: always BACKUP your database first PLUS test a restore

Thanks for your quick reply, i'll do tests with a cloned vm first :)
I have snapshoted the original too !
